Server/src/controllers/dataExtractionController.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response } from 'express';
+import { DataProcessService } from '../services/dataProcess/dataProcessService'
+
+/**
+ * The dataExtractionController is responsible for preparing a request and extracting key information
+ * from the request. The controller also finds the extension of the input file as processing is done based on it
+ */
+
+export class dataExtractionController {
+  private filePathOfUpload: string;
+  private fileExtension: string
+
+  constructor(filePath: string, fileExtension: string) {
+    this.filePathOfUpload = filePath;
+    this.fileExtension = fileExtension
+  }
+
+  createRequest(request: Request, response: Response) {
+    if (!request.body) {
+      response.status(400).send({
+        message: "Request Body is empty."
+      });
+    }
+    else {
+      //let fileName: any = request.file.originalname as any;
+      let extension = this.fileExtension.split('.').pop();
```

Comment:
why do you need to extract extension from a variable called fileExtension? Either extract it in constructor, or name the variable "fileName"